src/main/java/com/openlattice/chronicle/services/surveys/SurveysService.java

```diff
@@ -436,4 +444,547 @@ public void submitAppUsageSurvey(
+        EntityKey participatedInEK = getParticipatedInEntityKey( participatedInESID, participatedInEntityData );
+        entitiesByEK.put( participatedInEK, participatedInEntityData );
+
+        EntityKey participantEK = getParticipantEntityKey( participantESID, participantId );
+
+        edgesByEntityKey.add( Triple.of( participantEK, participatedInEK, surveyEK ) );
+
+        // edge: person -> respondswith -> submission
+
+        Map<UUID, Set<Object>> submissionEntityData = getSubmissionEntity( studyId, participantId, dateTime );
+        EntityKey submissionEK = getSubmissionEntityKey( submissionESID, submissionEntityData );
+        submissionEntityData.remove( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ) ); // shouldn't be stored
+        submissionEntityData.remove( edmCacheManager.getPropertyTypeId( OL_ID_FQN ) ); // shouldn't be stored
+        entitiesByEK.put( submissionEK, submissionEntityData );
+
+        Map<UUID, Set<Object>> respondsWithSubmissionEntity = getRespondsWithSubmissionEntity( studyId,
+                participantId,
+                dateTime );
+        EntityKey respondsWithEK = getRespondsWithEntityKey( respondsWithESID, respondsWithSubmissionEntity );
+        respondsWithSubmissionEntity
+                .remove( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ) ); // shouldn't be stored
+        entitiesByEK.put( respondsWithEK, respondsWithSubmissionEntity );
+
+        edgesByEntityKey.add( Triple.of( participantEK, respondsWithEK, submissionEK ) );
+
+        surveyData
+                .forEach( entity -> {
+                    // edge: question -> partof -> survey
+                    Map<UUID, Set<Object>> questionEntityData = getQuestionEntityData( entity );
+                    EntityKey questionEK = getQuestionEntityKey( questionESID, questionEntityData );
+                    entitiesByEK.put( questionEK, questionEntityData );
+
+                    Map<UUID, Set<Object>> questionPartOfSurveyEntity = getQuestionPartOfEntity( entity );
+                    EntityKey partOfEK = getQuestionPartOfSurveyEntityKey( partOfEntitySetId,
+                            questionPartOfSurveyEntity );
+                    entitiesByEK.put( partOfEK, questionPartOfSurveyEntity );
+
+                    edgesByEntityKey.add( Triple.of( questionEK, partOfEK, surveyEK ) );
+
+                    //edge: answer -> registeredfor -> timerange
+                    Map<UUID, Set<Object>> timeRangeEntityData = getTimeRangeEntity( entity );
+                    EntityKey timeRangeEK = getTimeRangeEntityKey( timeRangeESID, timeRangeEntityData );
+                    entitiesByEK.put( timeRangeEK, timeRangeEntityData );
+                } );
+
+        return new EntitiesAndEdges( entitiesByEK, edgesByEntityKey );
+    }
+
+    private Set<DataEdgeKey> getDataEdgeKeysFromEntityKeys(
+            Set<Triple<EntityKey, EntityKey, EntityKey>> edgesByEntityKey,
```

Comment:
This seems common enough that it might be worth adding a data class for this structure just to be explicit about the meaning / ordering.